src/contracts/permissions/Pausable.sol

```diff
@@ -20,114 +19,105 @@ import "../interfaces/IPausable.sol";
  * @dev We note as well that we have chosen to identify flags by their *bit index* as opposed to their numerical value, so, e.g. defining `DEPOSITS_PAUSED = 3`
  * indicates specifically that if the *third bit* of `_paused` is flipped -- i.e. it is a '1' -- then deposits should be paused
  */
-contract Pausable is IPausable {
+abstract contract Pausable is IPausable {
+    /// Constants
+
+    uint256 internal constant _UNPAUSE_ALL = 0;
+
+    uint256 internal constant _PAUSE_ALL = type(uint256).max;
+
     /// @notice Address of the `PauserRegistry` contract that this contract defers to for determining access control (for pausing).
-    IPauserRegistry public pauserRegistry;
+    IPauserRegistry public immutable pauserRegistry;
+
+    /// Storage
 
-    /// @dev whether or not the contract is currently paused
+    /// @dev Do not remove, deprecated storage.
+    IPauserRegistry private __deprecated_pauserRegistry;
+
+    /// @dev Returns a bitmap representing the paused status of the contract.
     uint256 private _paused;
 
-    uint256 internal constant UNPAUSE_ALL = 0;
-    uint256 internal constant PAUSE_ALL = type(uint256).max;
+    /// Modifiers
 
-    /// @notice
+    /// @dev Thrown if the caller is not a valid pauser according to the pauser registry.
     modifier onlyPauser() {
```

Comment:
Should we make `onlyPauser` and `onlyUnpauser` internal instead to save a little bit more gas?
`onlyPauser` is used twice in this contract but not something we ever use in inheriting contracts in core so the readability of having it as a modifier not as important imo.

Reply:
i checked, these don't save us bytecode